lib/irb/cmd/subirb.rb

```diff
@@ -10,32 +10,35 @@
 #
 
 require_relative "nop"
-require_relative "../ext/multi-irb"
 
 module IRB
   # :stopdoc:
 
   module ExtendCommand
     class IrbCommand < Nop
       def execute(*obj)
+        require_relative "../ext/multi-irb"
         IRB.irb(nil, *obj)
       end
     end
 
     class Jobs < Nop
       def execute
+        require_relative "../ext/multi-irb"
         IRB.JobManager
       end
     end
 
     class Foreground < Nop
       def execute(key)
+        require_relative "../ext/multi-irb"
         IRB.JobManager.switch(key)
       end
     end
 
     class Kill < Nop
       def execute(*keys)
+        require_relative "../ext/multi-irb"
```

Comment:
The overhead of `require_relative` would be hopefully negligible, but it seems ugly that you need to repeat this for every command in this file, especially when this file itself is lazily loaded for non-`show_cmds` stuff.

Is it really important to avoid loading `../ext/multi-irb` on `show_cmds`? The implementation complexity by this seems to outweigh the benefit of not loading them if it's just about the initial loading speed and there's no side effect besides method definitions.

Reply:
No it's not for speed. It's because these extensions, like `multi-irb`, [overrides some top-level methods](https://github.com/ruby/irb/blob/master/lib/irb/ext/multi-irb.rb#L180-L262).
So if `show_cmds` loads it, the users would have IRB's core components/top-level methods patched without noticing it.

Reply:
I wouldn't call methods on `IRB` as top-level methods, but I agree that it's a surprising side effect.

The danger is hard to notice from reading the current code. Maybe worth defining a helper method for doing `require_relative "../ext/multi-irb"`, indicating its side effect from the method name?